File: qt_cam_w/views/camera_view.py
from PySide6.QtWidgets import QVBoxLayout, QWidget

from .camera import CameraWidget


class CameraView(QWidget):
    def __init__(self, stream_name: str, stream_type: int, parent=None) -> None:
        super().__init__(parent)
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)
        layout.setSpacing(0)
        self.camera_widget = CameraWidget(stream_name, stream_type, self)
        layout.addWidget(self.camera_widget)

File: qt_cam_w/views/camera_view.cc
#include "camera_view.h"

#include <QVBoxLayout>

#include "camera.h"

CameraView::CameraView(const QString &stream_name, int stream_type, QWidget *parent)
    : QWidget(parent) {
  auto *layout = new QVBoxLayout(this);
  layout->setContentsMargins(0, 0, 0, 0);
  layout->setSpacing(0);

  camera_widget = new CameraWidget(stream_name.toStdString(),
                                   static_cast<VisionType>(stream_type),
                                   this);
  layout->addWidget(camera_widget);
}

CameraView::~CameraView() = default;
